repos/system_upgrade/el8toel9/actors/checkvdo/libraries/checkvdo.py

```diff
@@ -0,0 +1,180 @@
+import os
+
+from leapp import exceptions
+from leapp import models
+from leapp import reporting
+from leapp.libraries import stdlib
+
+def _run_cmd(command, checked = True):
+    """ Run the specified command returning the result. """
+    try:
+        result = stdlib.run(command, checked = checked)
+    except stdlib.CalledProcessError:
+        raise exceptions.StopActorExecutionError('Failure to execute "{0}"'.format(' '.join(command)))
+    return result
+
+def _canonicalize_device_path(path):
+    return os.path.realpath(path)
+
+#
+# Methods used in discovering VDO devices which require migration to
+# lvm-based management.
+#
+def _get_unmigrated_vdo_blkid_results():
+    """ Collect devices that have not been migrated to lvm-based management. """
+    # VDO devices that have been created by or migrated to lvm-based management
+    # are not identified by blkid.
+    command = ['blkid', '--output', 'device', '--match-token', 'TYPE=vdo']
+    result = _run_cmd(command, checked = False)
+    exit_code = result['exit_code']
+    if (exit_code != 0) and (exit_code != 2):
+        raise exceptions.StopActorExecutionError('blkid exited with code: {0}'.format(exit_code))
```

Comment:
This should be handled more gracefully. For example with an inhibitor with more information about what went wrong. This is quite harsh.

Reply:
@vinzenz 
Addressed in commit 465a92f via generation of inhibitory report.